idb/postgres/postgres_migrations.go

```diff
@@ -99,27 +100,16 @@ func needsMigration(state MigrationState) bool {
 	return state.NextMigration < len(migrations)
 }
 
-// upsertMigrationStateTx updates the migration state, and optionally increments the next counter with an existing
-// transaction.
-func upsertMigrationStateTx(tx *sql.Tx, state *MigrationState, incrementNextMigration bool) (err error) {
+// upsertMigrationState updates the migration state, and optionally increments
+// the next counter with an existing transaction.
+// If `tx` is nil, use a normal query.
+func upsertMigrationState(db *IndexerDb, tx *sql.Tx, state *MigrationState, incrementNextMigration bool) error {
 	if incrementNextMigration {
 		state.NextMigration++
 	}
 	migrationStateJSON := encoding.EncodeJSON(state)
-	_, err = tx.Exec(setMetastateUpsert, migrationMetastateKey, migrationStateJSON)
 
-	return err
-}
-
-// upsertMigrationState updates the migration state, and optionally increments the next counter.
-func upsertMigrationState(db *IndexerDb, state *MigrationState, incrementNextMigration bool) (err error) {
-	if incrementNextMigration {
-		state.NextMigration++
-	}
-	migrationStateJSON := encoding.EncodeJSON(state)
-	_, err = db.db.Exec(setMetastateUpsert, migrationMetastateKey, migrationStateJSON)
-
-	return err
+	return db.setMetastate(tx, migrationMetastateKey, string(migrationStateJSON))
```

Comment:
This needs the weird post-success-next-migration-update thing because it might be used inside a retry function.